pkg/sql/pgwire/server.go

```diff
@@ -741,12 +767,19 @@ func (s *Server) ServeConn(ctx context.Context, conn net.Conn, socketType Socket
 // attacker) will not know if the cancellation attempt succeeded. Errors are
 // logged so that an operator can be aware of any possibly malicious requests.
 func (s *Server) handleCancel(ctx context.Context, conn net.Conn, buf *pgwirebase.ReadBuffer) {
+	var resp *serverpb.CancelQueryByKeyResponse
 	var err error
 	defer func() {
-		if err != nil {
+		telemetry.Inc(sqltelemetry.CancelRequestCounter)
```

Comment:
optional: i generally this `err` in defer pattern can be shadowed pretty easily. 

how about:

```go
resp, err := func() (*serverpb.CancelQueryByKeyResponse, error) {
backendKeyDataBits, err = buf.GetUint64()
	if err != nil {
		return nil, err
	}
	cancelKey := pgwirecancel.BackendKeyData(backendKeyDataBits)
	// The request is forwarded to the appropriate node.
	req := &serverpb.CancelQueryByKeyRequest{
		SQLInstanceID:  cancelKey.GetSQLInstanceID(),
		CancelQueryKey: cancelKey,
	}
	resp, err := s.execCfg.SQLStatusServer.CancelQueryByKey(ctx, req)
	if len(resp.Error) > 0 {
		err = errors.CombineErrors(err, errors.Newf("error from CancelQueryByKeyResponse: %s", resp.Error))
	}
	return resp, err
}()

telemetry.Inc(sqltelemetry.CancelRequestCounter)
s.metrics.PGWireCancelTotalCount.Inc(1)
		if resp != nil && resp.Canceled {
			s.metrics.PGWireCancelSuccessfulCount.Inc(1)
		} else if err != nil {
			if status := status.Convert(err); status.Code() == codes.ResourceExhausted {
				s.metrics.PGWireCancelIgnoredCount.Inc(1)
			}
			log.Sessions.Warningf(ctx, "unexpected while handling pgwire cancellation request: %v", err)
		}
return resp, err
```